### DISCUSSION TITLE
TD de GitHub

### DISCUSSION THREAD

**Discussion by BWhannou** [ML-for-B-E/.github#10] [General]
@akouete-kpakpo Hello Kevin! j'ai discuté avec Espéran à propos du TD GitHub. Il pense que ce serait mieux qu'on fasse sans. Toujours pour simplifier. 

Une solution serait que sur les PC BE, on installe tous les dossiers avant le début de l'école 🤷

**Comment by akouete-kpakpo** [comment]
Ok je comprends le besoin de simplification. 

La question n'est pas tant que ça l'installation mais plutôt le versionnement et le débogage. Car si les élèves touchent aux packages privés qui seront éditables et obtiennent des erreurs, le suivi Git nous permettra de tracer plus facilement les diffs et de déboguer plus rapidement.

**Reply by akouete-kpakpo** [reply]
mais je suis d'accord que ce n'est pas très cohérent de le mettre dans la semaine ML/Stats. On pourrait le faire pendant la semaine Info car:
- c'est plus cohérent
- ils ont beaucoup de temps en prenant en compte la pré-rentrée

**Reply by akouete-kpakpo** [reply]
@BWhannou

**Reply by akouete-kpakpo** [reply]
et aussi notre capacité à réagir. Si on détecte une erreur chez un élève et qu'on fait un bugfix qu'on veut rendre rapidement disponible à l'ensemble de la classe, c'est possible avec git.

**Reply by chrichri17** [reply]
> ils ont beaucoup de temps en prenant en compte la pré-rentrée

Ceci n'est pas vrai. N'oublions pas que tout le monde ne participe pas à la pré-rentrée. Et d'après les retours de Fidèle, la pré-rentrée n'a pas vraiment facilité la tâche lors de la première semaine.

**Comment by BWhannou** [comment]
Ok, understood. For me the objective of the GitHub TP would just be to understand `git clone` in order to do the [manual](https://github.com/ML-for-B-E/.github/wiki/ML-Summer-Camp-courses) , I will let you, @akouete-kpakpo , coordinate with @ML-for-B-E/info

**Reply by akouete-kpakpo** [reply]
Not only `git clone` but also `git status` and `git pull`.

Understand the vocabulary of git: a commit, a branch.

Thanks we'll discuss of it with @fideledegni and @chrichri17

**Reply by chrichri17** [reply]
I think, and I agree with Espéran, that talking about git and using it is not a priority. Participants will get lost for sure at some point. And we don't have that much time either. (I remember my git training with Brunice and Bassimath took about 2h30. And I assume for 150 participants, it will be more).

There are a bunch of alternatives, although they are not suited to debugging, fixing, etc. (this one for example https://docs.python.org/3/library/http.server.html). We can also make our materials suitable to this system.

In summary, I think we can use this time allocated to a git training to do more exercises in python, ML or whatever 😅

**Reply by akouete-kpakpo** [reply]
Well @chrichri17  I think 2 h is enough to review 
- git clone 
- git status 
- git pull

We'll stick to git for versionning and debugging.

But for the training course, let's say I'll just cover it quickly at the beginning of my courses.
